execution.bs

```diff
@@ -4108,28 +4111,219 @@ namespace std::execution {
+
+    <pre highlight=c++>
+    template &lt;class T>
       concept <i>movable-value</i> = <i>// exposition only</i>
         move_constructible&lt;decay_t&lt;T>> &&
         constructible_from&lt;decay_t&lt;T>, T>;
     </pre>
 
+    For function types `F1` and `F2` denoting `R1(Args1...)` and `R2(Args2...)`
+    respectively, <code><i>MATCHING-SIG</i>(F1, F2)</code> is `true` if and only if 
+    `same_as<R1(Args&&...), R2(Args2&&...)>` is `true`.
+
+## Asynchronous functions <b>[async.func]</b> ## {#spec-execution-async.func}
+
+Subsections [exec.queries] through [exec.snd] specify a set of interrelated concepts
+for defining asynchronous algorithms. This subsection defines some terms and then
+specifies the [=async function protocol=] that asynchronous algorithms must
+adhere to.
+
+### Terms and definitions <b>[async.defn]</b> ### {#spec-execution-async.defn}
+
+Later sections define the following terms in detail. Below we list them and the
+role they play in the async function protocol.
+
+: query object
+:: A customization point object ([customization.point.object]) that accepts a
+    *queryable* as its first argument and whose call expressions model
+    `regular_invocable` ([concept.regularinvocable]). Query objects impose
+    syntactic and semantic requirements on their return types (which cannot be
+    `void`).
+
+: queryable
+:: A type that responds to zero or more queries. A queryable object ensures
+    that the results of its queries do not depend on the *cv* qualification
+    of the object's type, nor on its value category.
+
+: environment
+:: A queryable object obtainable from a receiver that contains runtime
+    configuration information local to the current operation.
+
+: execution resource
+:: An object, like a thread or a thread pool, that is capable of executing
+    code, and onto which asynchronous operations can be scheduled for
+    execution.
+
+: receiver
+:: An object that receives the results of an asynchronous operation when it
+    completes. It can receive the values on success, an error on failure, or
+    a stopped notification when the operation is cancelled. Additionally,
+    the receiver provides an environment that describes the properties of
+    the 
+
+: operation state
+:: An object responsible for holding the state associated with an asynchronous
+    operation while it is running. That state includes the receiver and the
+    operation states of any nested operations. Operation states are the
+    result of connecting senders and receivers, but they are properly thought
+    of as part of the sender's implementation. Operation states need to be
+    explicitly started.
+
+: sender
+:: An object that encapsulates the description of asynchronous operation,
+    independent of where it will begin execution and what will happen on
+    completion.
+
+: scheduler
+:: A handle to an execution resource that presents a uniform, generic
+    interface for scheduling work onto that resource.
```

Comment:
An object that abstracts an execution resource to... Multiple scheduler may refer to the same execution resource.